bacon: add page (#15931)
tldr-pages/tldr: pages/common/bacon.md

# bacon

> A background code checker for Rust.
> More information: <https://github.com/Canop/bacon>.

- Run `cargo check` whenever a change is detected in the current directory:

`bacon`

> - Run `cargo run` whenever a change is detected in the current directory:
> 
> `bacon run`

Reviewer: > I also did not add in every command, as they are only minor variations of the 2nd example listed, e.g. `bacon doc`, `bacon clippy`. Should I include those?

Maybe include them in a single example?

```suggestion
- Run specific job whenever a change is detected in the current directory:

`bacon {{run|test|clippy|doc|...}}`
```

Author: Ah, of course. I'll add an example for `--list-jobs` too

> - Run `cargo test` whenever a change is detected in the given directory:
> 
> `bacon test --path {{path/to/directory}}`

Reviewer: This doesn't seem to work:
```
$ bacon test --path somedirectory 
error: unexpected argument '--path' found

  tip: a similar argument exists: '--watch'
  tip: to pass '--path' as a value, use '-- --path'

Usage: bacon <--help|--version|--prefs|--headless|--summary|--no-summary|--wrap|--no-wrap|--reverse|--no-reverse|--help-line|--no-help-line|--list-jobs|--completion-list-jobs|--offline|--init|--job <job>|--no-default-features|--features <features>|--all-features|--export-locations|--no-export-locations|--watch <watch>|--project <project>|--config-toml <CONFIG_TOML>|ARGS|ADDITIONAL_JOB_ARGS>
```

It does work without `--path` though.

```suggestion
- Run `cargo test` whenever a change is detected in the given directory:

`bacon test {{path/to/directory}}`
```

Author: Yeah looks like the readme had the (I'm guessing) old flag, though `--watch` doesn't seem to work either. Just the path works, but only if you provide it with the relative / absolute path i.e. `./path` not `path`

> - Run against all targets whenever a change is detected in the current directory:

Reviewer: ```suggestion
- Run `cargo check` against all targets whenever a change is detected in the current directory:
```

> `bacon --job check-all`

Reviewer: This also works without `--job`.
```suggestion
`bacon check-all`
```

# bacon

> A background code checker for Rust.
> More information: <https://github.com/Canop/bacon>.

- Run `cargo check` whenever a change is detected in the current directory:

`bacon`

- Run `cargo test` whenever a change is detected in the given directory:

`bacon test {{path/to/directory}}`

- Run `cargo check` against all targets whenever a change is detected in the current directory:

`bacon check-all`

> - Run specific job whenever a change is detected in the current directory:

Reviewer: ```suggestion
- Run a specific job whenever a change is detected in the current directory:
```
